moonstone/plot/functions.py

```diff
@@ -0,0 +1,88 @@
+import sys
+import typing
+from typing import List
+import warnings
+
+
+def check_list_of(listx, expectedtype):    # expectedtype can be tuple of type
+    """
+    check the type of what's in listx
+    """
+    return all(isinstance(s, expectedtype) for s in listx)
+
+
+def check_type(to_check, expectedtype: type):
+    """
+    check if type of 'to_check' is right
+    """
+    # type(List[str]) = typing._GenericAlias in python3.7 and typing.GenericMeta in python3.6
```

Comment:
Careful here, typing and type are two different things. [Pydantic](https://pydantic-docs.helpmanual.io/) is a nice library that performs validations based on typing.

Here you end up check the type of the type and it gets a bit confusing in the end.